src/Microsoft.VisualStudio.Editor.Razor/DefaultImportDocumentManager.cs

```diff
@@ -0,0 +1,158 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using Microsoft.AspNetCore.Razor.Language;
+using Microsoft.CodeAnalysis.Razor;
+
+namespace Microsoft.VisualStudio.Editor.Razor
+{
+    internal class DefaultImportDocumentManager : ImportDocumentManager
+    {
+        private readonly FileChangeTrackerFactory _fileChangeTrackerFactory;
+        private readonly ForegroundDispatcher _foregroundDispatcher;
+        private readonly ErrorReporter _errorReporter;
+        private readonly RazorTemplateEngineFactoryService _templateEngineFactoryService;
+        private readonly Dictionary<string, ImportTracker> _importTrackerCache;
```

Comment:
Minor: I personally prefer taking dependencies on interfaces. So this would be "IDictionary<string, ImportTracker>" in this case

Reply:
The only problem with using the interface variants for private fields is that it limits you to what APIs you can use when making modifications in the future. By not using the interface here it's a bet that we'll more often than not be using the field in some manner; basically i'll have more APIs to make use of and wont have to go searching.

If I were to use the interface I'd be betting on us needing to change the type in the future which is typically a less common scenario.